src/panels/lovelace/editor/config-elements/hui-grid-card-editor.ts

```diff
@@ -49,35 +50,18 @@ const SCHEMA = [
 
 @customElement("hui-grid-card-editor")
 export class HuiGridCardEditor extends HuiStackCardEditor {
+  protected _schema: readonly HaFormSchema[] = SCHEMA;
+
   public setConfig(config: Readonly<GridCardConfig>): void {
     assert(config, cardConfigStruct);
     this._config = config;
   }
 
-  protected render() {
-    if (!this.hass || !this._config) {
-      return nothing;
-    }
-
-    const data = { square: true, ...this._config };
-
-    return html`
-      <ha-form
-        .hass=${this.hass}
-        .data=${data}
-        .schema=${SCHEMA}
-        .computeLabel=${this._computeLabelCallback}
-        @value-changed=${this._valueChanged}
-      ></ha-form>
-      ${super.render()}
-    `;
-  }
-
-  private _valueChanged(ev: CustomEvent): void {
-    fireEvent(this, "config-changed", { config: ev.detail.value });
+  protected formData(): object {
+    return { square: true, ...this._config };
   }
 
-  private _computeLabelCallback = (schema: SchemaUnion<typeof SCHEMA>) =>
+  protected _computeLabelCallback = (schema: SchemaUnion<typeof SCHEMA>) =>
```

Comment:
I thought that probably I should not have had to define this function in the grid card editor, and just defer to the function in the superclass, but when I tried to type this as `SchemaUnion<ReturnType<typeof this._schema>>`, I couldn't get around the "type instantiation is infinitely deep error". 

So this is the workaround, to use `SCHEMA` instead of `this._schema`.

Reply:
I think it's ok like this 🙂